consensus/polybft/state_sync_manager.go

```diff
@@ -399,14 +412,33 @@ func (s *stateSyncManager) PostEpoch(req *PostEpochRequest) error {
 }
 
 // PostBlock notifies state sync manager that a block was finalized,
-// so that it can build state sync proofs if a block has a commitment submission transaction
+// so that it can build state sync proofs if a block has a commitment submission transaction.
+// Additionally, it will remove any processed state sync events and their proofs from the store.
 func (s *stateSyncManager) PostBlock(req *PostBlockRequest) error {
+	events, err := s.eventsGetter.getEventsFromReceipts(req.FullBlock.Block.Header, req.FullBlock.Receipts)
```

Comment:
This would mean we will only remove executed state syncs from this block. If an error occurs while deleting them, or if `eventsGetter` returns an error, executed state syncs from this block will never be deleted. We need that retry mechanism that stake manager and checkpoint manager have, but maybe it doesn't matter, because, this PR changes all that: https://github.com/0xPolygon/polygon-edge/pull/1972.

Your call.

Reply:
This is similar to the comment regarding logging or returning error.

Regarding the `eventsGetter` failure, do we have a reason to believe that, if parsing of logs for some block failed it could be successful on retry to parse that same log?

Reply:
`Regarding the eventsGetter failure, do we have a reason to believe that, if parsing of logs for some block failed it could be successful on retry to parse that same log` - if you are only using `getEventsFromReceipts` then no. If we are to implement the retry mechanism, than yes.

`t: Current mechanism in checkpoint manager, which uses eventsGetter.getFromBlocks() doesn't work well, since last processed block is saved after each PostBlock()` - this is not true. It does save the last processed block in `PostBlock`, but it only saves it after `eventsGetter` successfully returns the events from given blocks, and after they are saved in `db` successfully.

This PR https://github.com/0xPolygon/polygon-edge/pull/1972 goes even further, and introduces the concept of global db transaction, so if one component fails to update its state, all of them fail, and nothing will be stored in db.